Name: Hide Quick Launch Bar
Version: 1.3.1-2
Author: Jason Robitaille
Description: Hides the quick launch bar (while still alowing the quick launch wave)  (MIT license)

--- .orig/usr/lib/luna/system/luna-applauncher/app/models/SystemManagerService.js
+++ /usr/lib/luna/system/luna-applauncher/app/models/SystemManagerService.js
@@ -7,7 +7,7 @@
 SystemManagerService.showQuickLaunch = function(show) {
 	var request = new Mojo.Service.Request(SystemManagerService.SysMgrId, {
 		method: 'systemUi',
-		parameters: {'quicklaunch':show}
+		parameters: {'quicklaunch':false}
 	});
 	request = null;
 }
\ No newline at end of file
--- .orig/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
+++ /usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
@@ -8,7 +8,7 @@
 	 * hardcoding the quick launch height until we can find a better way. 
 	 * This value = ql graphic height - the offset from the top (small fade area)
 	 */
-	kQuickLaunchHeight: 67,
+	kQuickLaunchHeight: 0,
 	
 	kPageIndicatorSpacing: 6, 	/* spacing between each page indicator */
 	kPageMargin: 10, /* 10 pixel margin on each side of a page */
--- .orig/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ /usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -17,6 +17,11 @@
 		this.wifiState = 'Off';	
 		this.wifiRadio = false;	
 		
+		this.controller.serviceRequest('palm://com.palm.systemmanager', {
+			method: 'systemUi',
+			parameters: {"quicklaunch": false}
+		});
+		
 		$('rssi').className = 'nobars';
 		$('datanet').className = 'off';
 		$('wifi').className = 'off';
@@ -236,6 +241,13 @@
 			}
 			$('appname').innerHTML = $L("Preferences");
 		}
+		else if (payload.id == "com.palm.launcher"){
+			this.controller.serviceRequest('palm://com.palm.systemmanager', {
+				method: 'systemUi',
+				parameters: {"quicklaunch": false}
+			});
+			$('appname').innerHTML = payload.appmenu || payload.title;
+		}
 		else 
 			$('appname').innerHTML = payload.appmenu || payload.title;
 		$('appname').show();
@@ -242,6 +254,10 @@
 		this.foregroundAppID = payload.id;
 	}
 	else{
+		this.controller.serviceRequest('palm://com.palm.systemmanager', {
+			method: 'systemUi',
+			parameters: {"quicklaunch": false}
+		});
 		$('appname').hide();
 		$('carrier').show();
 		this.foregroundAppID=null;
